include/aspect/simulator_access.h

```diff
@@ -241,6 +241,13 @@ namespace aspect
       void
       get_artificial_viscosity(Vector<float> &viscosity_per_cell) const;
 
+      /**
+       * Returns the entropy viscosity on each locally owned cell as it is
+       * used to stabilize the composition equation.
+       */
+      void
+      get_artificial_viscosity_composition(Vector<float> &viscosity_per_cell,
+                                           const unsigned int compositional_variable) const;
```

Comment:
The place where this would be used is e.g. in a visualization postprocessor or mesh refinement plugin where it is much easier to just hand over the index of the compositional field than to get the AdvectionField... or did I misunderstand the question?